e2e/e2e_test.go

```diff
@@ -99,6 +99,50 @@ func TestE2EBasic(t *testing.T) {
 	_ = cleanup()
 }
 
+// TestE2EArgumentParsing runs the CTS binary and checks to ensure that
+// argument parsing works properly for both `=` and space-delimited argument values.
+// For example, both `--a=b` and `--a b` should be acceptable ways to specify args.
+func TestE2EArgumentParsing(t *testing.T) {
+	srv := newTestConsulServer(t)
+	defer srv.Stop()
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "cmd_parsing")
+	cleanupTemp := testutils.MakeTempDir(t, tempDir)
+
+	// Create an extra, empty directory that will be searched for config files.
+	// This is done because the `StartCTS` function always includes its own
+	// argument of `-config-file=` and we cannot reference the same files twice.
+	emptyDir := fmt.Sprintf("%s%s", tempDirPrefix, "cmd_parsing_emptydir")
+	cleanupEmpty := testutils.MakeTempDir(t, emptyDir)
+	defer cleanupEmpty()
```

Comment:
`cleanupTemp()` and `cleanupEmpty()` are called in two different ways, one is in a `defer` here, the other is in `_ = cleanupTemp()` at the end of the test. Is that intentional?

Reply:
Yeah, this was mainly a clone of the `TestE2EBasic` function. It has:
```
	cleanup := testutils.MakeTempDir(t, tempDir)
	// no defer to delete directory: only delete at end of test if no errors
```
for its cleanup approach, which allowed developers to debug by looking at the directory in the event that the configuration was bad. I was replicating that behavior, but since we're using the same config file as the basic test, I can likely just defer it and not worry about need to debug in that way.